kokoro/scripts/test/go_test.Dockerfile

```diff
@@ -14,7 +14,7 @@
 
 # To edit this file, follow these instructions: go/sdi-integ-test#updating-the-test-runner-container.
 
-FROM golang:1.22-bullseye
+FROM golang:1.23-bullseye
```

Comment:
Following steps mentioned above (go/sdi-integ-test#updating-the-test-runner-container) to update this file.

Reply:
Done in : https://github.com/GoogleCloudPlatform/ops-agent/pull/1927